microprofile.ls/README.md

```diff
@@ -43,8 +46,52 @@ Building the Language Server
 Clients
 -------
 
-Here are some clients consuming the Quarkus language server:
+Here are some clients consuming the MicroProfile language server:
 
  * Eclipse with [quarkus-lsp4e (POC)](https://github.com/angelozerr/quarkus-lsp4e)
  * IntelliJ with [intellij-quarkus](https://github.com/jeffmaury/intellij-quarkus)
- * Visual Studio Code with [vscode-quarkus](https://github.com/redhat-developer/vscode-quarkus)
\ No newline at end of file
+ * Visual Studio Code with [vscode-quarkus](https://github.com/redhat-developer/vscode-quarkus)
+ 
+Code Snippets
+-------
+
+Java and properties completion snippets are managed by the MicroProfile LS (snippets on server side) by using Java SPI.
+
+## Describes snippets in JSON
+
+Snippets are described in a JSON files by following the [vscode snippet format](https://code.visualstudio.com/docs/editor/userdefinedsnippets#_create-your-own-snippets).
+
+A context can be described to show the snippet according to a condition:
+
+ * for properties file: the context/extension can be declared in the snippet to show the snippet only if the extension belongs to the project.
+
+```json
+"Add datasource properties": {
+		"prefix": "qds",
+		"body": [
+			...
+		],
+		"context": {
+			"extension": "quarkus-agroal"
+		}
+	}
+```
+
+means that the snippet is shown only if the project have the `quarkus-agroal` dependency.
+    
+ * for java files (not available for the moment)
+ 
+## Add new internal snippets
+
+To register a snippet you need to add it in :
+
+ * [MicroProfileJavaSnippetRegistryLoader](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/snippets/MicroProfileJavaSnippetRegistryLoader.java) if the new snippets is for java files. 
+ * [MicroProfilePropertiesSnippetRegistryLoader](https://github.com/redhat-developer/quarkus-ls/blob/master/microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/snippets/MicroProfilePropertiesSnippetRegistryLoader.java) if the new snippets is for properties files.
+
+## Add new external snippets
+
+To add external snippets (like Quarkus) you need to create an implementation of ISnippetRegistryLoader and register it with Java SPI. See for Quarkus:
```

Comment:
To add external snippets (like Quarkus snippets) an implementation of `ISnippetRegistryLoader` and register it with Java SPI must be created. See for Quarkus snippets: